microcosm_flask/metrics_classifier.py

```diff
@@ -14,8 +14,17 @@ class StatusCodeClassifier(Classifier):
     """
     def label_result(self, result):
         _, status_code, _ = parse_response(result)
-        return str(status_code)
+        return self.normalize_status_code(status_code)
 
     def label_error(self, error):
         status_code = extract_status_code(error)
-        return str(status_code)
+        return self.normalize_status_code(status_code)
+
+    # Return only the first digit of the status code followed by xx as a string
+    # Example: 403 becomes "4xx"
+    # This will reduce the cardinality of metrics collected which will reduce cost
+    # For detailed status code errors we have logs
+    def normalize_status_code(self, status_code: int) -> str:
+        if isinstance(status_code, int):
```

Comment:
This type-hint is redundant. If you *are* getting multiple types, that should be reflected in the signature above.

Reply:
just being over-safe, as this is a library.
But I agree, it can be removed.

Reply:
It is a library, but how it's getting the status code is from `parse_response`, another internal function.

We really should have more typehints in this repo, but the expected values of the responses should mean you'd always get an int here.